crates/e2e/tests/e2e/services.rs

```diff
@@ -187,12 +188,20 @@ impl OrderbookServices {
             price_estimator.clone(),
             order_validator,
         ));
+
+        let (shutdown_sender, shutdown_receiver) = tokio::sync::oneshot::channel::<()>();
         orderbook::serve_api(
             db.clone(),
             orderbook,
             quoter,
             API_HOST[7..].parse().expect("Couldn't parse API address"),
-            pending(),
+            async {
+                if shutdown_receiver.await.is_err() {
+                    // Tests dropping the shutdown_sender would terminate the service by accident.
+                    // Let's wait for an explicit shutdown message before terminating the service.
+                    std::future::pending::<()>().await;
```

Comment:
The pending future we had before never got dropped. The problem started happening after I started using the oneshot channel for the shutdown mechanism.
Just replacing the `pending()` future with a future waiting for a message from a oneshot channel caused the future to terminate once the channel sender has been dropped. This meant all the existing e2e tests would have to keep the `shutdown_sender` alive. To not make every test manage the `shutdown_sender` when they never had the intention to terminate the service, I required an actual shutdown message to be sent (which can be differentiated from termination caused by dropping the sender). If `shutdown_receiver.await` terminated because the sender got dropped, it was one of the tests which didn't want to terminate the service anyway so I used `pending()` as the expected fallback behavior.
But that doesn't matter anymore anyway because most of the problems can be solved by replacing the e2e with an unit test altogether.

Reply:
I see, my bad was that I didn't realise that if shutdown_receiver.await.is_ok() we also resolve the future.